fs/direntries_test.go

```diff
@@ -0,0 +1,97 @@
+package fs
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/ncw/rclone/fs/hash"
+	"github.com/stretchr/testify/assert"
+)
+
+type FakeObject string
```

Comment:
You should be able to use `fstest/mockobject` here I think rather than build your own FakeObject

Reply:
If it is giving you a circular import then change the package to `fs_test`

Reply:
Yes. I've got circular imports when I tried to use mockobject and mockdir. 
Thanks I'll change it.

Reply:
Fixed